src/components/Personalization/in-app-message-actions/actions/displayIframeContent.js

```diff
@@ -148,60 +87,145 @@ export const createIframe = (htmlContent, clickHandler) => {
-  element.className = `${ELEMENT_TAG_CLASSNAME}`;
-  Object.assign(
-    element.style,
-    buildStyleFromParameters(mobileParameters, webParameters)
-  );
+const displayBasedOnWebParams = (iframe, webParameters, container, overlay) => {
+  const { style: iframeStyle = {} } = webParameters[ALLOY_IFRAME_ID];
+  const {
+    style: messagingStyle = {},
+    params: messagingParams = {}
+  } = webParameters[ALLOY_MESSAGING_CONTAINER_ID];
 
-  return element;
+  const {
+    style: overlayStyle = {},
+    params: overlayParams = {}
+  } = webParameters[ALLOY_OVERLAY_CONTAINER_ID];
+
+  assign(iframe.style, iframeStyle);
+  assign(container.style, messagingStyle);
+
+  if (overlayParams.enabled) {
+    assign(overlay.style, overlayStyle);
+    document.body.appendChild(overlay);
+    document.body.style.overflow = "hidden";
+  }
+
+  const {
+    paramElement = "body",
+    insertionMethod = "appendChild"
+  } = messagingParams;
+
+  const element = document.querySelector(paramElement);
+  element[insertionMethod](container);
 };
 
-export const createOverlayElement = parameter => {
-  const element = document.createElement("div");
-  const backdropOpacity = parameter.backdropOpacity || 0.5;
-  const backdropColor = parameter.backdropColor || "#FFFFFF";
-  element.id = OVERLAY_TAG_ID;
-  element.className = `${OVERLAY_TAG_CLASSNAME}`;
+export const buildStyleFromMobileParameters = mobileParameters => {
+  const {
+    verticalAlign,
+    width,
+    horizontalAlign,
+    backdropColor,
+    height,
+    cornerRadius,
+    horizontalInset,
+    verticalInset,
+    uiTakeover = false
+  } = mobileParameters;
+
+  const style = {
+    width: width ? `${width}%` : "100%",
+    backgroundColor: backdropColor || "rgba(0, 0, 0, 0.5)",
+    borderRadius: cornerRadius ? `${cornerRadius}px` : "0px",
+    border: "none",
+    position: uiTakeover ? "fixed" : "relative",
+    overflow: "hidden"
+  };
+  if (horizontalAlign === "left") {
+    style.left = horizontalInset ? `${horizontalInset}%` : "0";
+  } else if (horizontalAlign === "right") {
+    style.right = horizontalInset ? `${horizontalInset}%` : "0";
+  } else if (horizontalAlign === "center") {
+    style.left = "50%";
+    style.transform = "translateX(-50%)";
+  }
+
+  if (verticalAlign === "top") {
+    style.top = verticalInset ? `${verticalInset}%` : "0";
+  } else if (verticalAlign === "bottom") {
+    style.position = "fixed";
+    style.bottom = verticalInset ? `${verticalInset}%` : "0";
+  } else if (verticalAlign === "center") {
+    style.top = "50%";
+    style.transform = `${
+      horizontalAlign === "center" ? `${style.transform} ` : ""
+    }translateY(-50%)`;
+    style.display = "flex";
+    style.alignItems = "center";
+    style.justifyContent = "center";
+  }
+
+  if (height) {
+    style.height = `${height}vh`;
+  } else {
+    style.height = "100%";
+  }
+  return style;
+};
 
-  Object.assign(element.style, {
+export const mobileOverlay = (overlay, mobileParameters) => {
```

Comment:
You can refactor the mobile display code to use the same `renderElement` function i described above.

The result is less code and fewer code paths with code duplication.

Reply:
imagine if your code that translates mobileParameters simply translates mobileParameteters to a valid `webParameters` object.  that way you can have maximum code reuse becuase the render function will always expect `webParameters` whether it came from konductor directly or was constructed based on `mobileParameters`

Reply:
Done